src/sentry/static/sentry/app/views/settings/components/settingsLayout.tsx

```diff
@@ -14,77 +18,182 @@ type Props = {
-  const childRoutes = childProps.routes || routes || [];
-  const childRoute = childProps.route || route || {};
-  return (
-    <React.Fragment>
-      <SettingsColumn>
-        <SettingsHeader>
-          <HeaderContent>
-            <StyledSettingsBreadcrumb
-              params={params}
-              routes={childRoutes}
-              route={childRoute}
-            />
-            <SettingsSearch routes={routes} router={router} params={params} />
-          </HeaderContent>
-        </SettingsHeader>
-
-        <MaxWidthContainer>
-          {typeof renderNavigation === 'function' && (
-            <SidebarWrapper>{renderNavigation()}</SidebarWrapper>
-          )}
-          <Content>{children}</Content>
-        </MaxWidthContainer>
-      </SettingsColumn>
-    </React.Fragment>
-  );
+type State = {
+  navVisible: boolean;
+  navOffsetTop: number;
+};
+
+class SettingsLayout extends React.Component<Props, State> {
+  static propTypes = {
+    renderNavigation: PropTypes.func,
+    route: PropTypes.object,
+    router: PropTypes.object,
+    routes: PropTypes.array,
+  };
+
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      /**
+       * This is used when the screen is small enough that the navigation should
+       * be hidden. On large screens this state will end up unused.
+       */
+      navVisible: false,
+      /**
+       * Offset mobile settings navigation by the height of main navigation,
+       * settings breadcrumbs and optional warnings.
+       */
+      navOffsetTop: 0,
+    };
+
+    this.headerRef = React.createRef();
+
+    // Close the navigation when navigating.
+    this.unlisten = browserHistory.listen(() => this.toggleNav(false));
+  }
+
+  componentWillUnmount() {
+    this.unlisten();
+  }
+
+  unlisten: () => void;
+  headerRef: React.RefObject<HTMLDivElement>;
+
+  toggleNav(navVisible: boolean) {
+    // when the navigation is opened, body should be scroll-locked
+    this.toggleBodyScrollLock(navVisible);
+
+    this.setState({
+      navOffsetTop: this.headerRef.current?.getBoundingClientRect().bottom ?? 0,
+      navVisible,
+    });
+  }
+
+  toggleBodyScrollLock(lock: boolean) {
+    const bodyElement = document.getElementsByTagName('body')[0];
+    window.scrollTo(0, 0);
+    bodyElement.style.overflow = lock ? 'hidden' : 'visible';
+    bodyElement.style.position = lock ? 'fixed' : 'static';
+    bodyElement.style.width = lock ? '100%' : 'auto';
+  }
+
+  render() {
+    const {params, routes, route, router, renderNavigation, children} = this.props;
+    const {navVisible, navOffsetTop} = this.state;
+
+    // We want child's view's props
+    const childProps =
+      children && React.isValidElement(children) ? children.props : this.props;
+    const childRoutes = childProps.routes || routes || [];
+    const childRoute = childProps.route || route || {};
+    const shouldRenderNavigation = typeof renderNavigation === 'function';
+
+    return (
+      <React.Fragment>
```

Comment:
You don't need `React.Fragment` here, because you are already wrapping the components using `SettingsColumn`😉

Reply:
Good catch!